src/Sema.zig

```diff
@@ -20583,7 +20583,12 @@ fn validateExternType(
             8, 16, 32, 64, 128 => return true,
             else => return false,
         },
-        .Fn => return !Type.fnCallingConventionAllowsZigTypes(ty.fnCallingConvention()),
+        .Fn => switch (ty.fnCallingConvention()) {
+            // For now we want to authorize PTX kernel to use zig objects, even if we end up exposing the ABI.
+            // The goal is to experiment with more integrated CPU/GPU code.
+            .PtxKernel => return true,
+            else => return !Type.fnCallingConventionAllowsZigTypes(ty.fnCallingConvention()),
+        },
```

Comment:
Why not just change `fnCallingConventionAllowsZigTypes` to return true for `.PtxKernel`?

Reply:
so `fnCallingConventionAllowsZigTypes` is already returning `true` for `.PtxKernel`. But that's the issue. Because I still want to be able to export the kernels. So we allow exporting functions that don't have Zig types in their CC, and we also allow exporting PtxKernel even if they have Zig types.